A3 review

lnxblog opened the pull request.

Commit:
Use assignment3 container image

Commit:
Remove full test for assignment 4

Full test for assignment 4 is completed through the buildroot
repository, this repo is only used for unit testing in assignment
4.

dwalkes committed:
Add threading tets for assignment 4

dwalkes committed:
Assignment 5 changes

dwalkes committed:
Changes for assignment 6

dwalkes committed:
Assignment 7 source

Implementation of assignment 7 and shared source stubs.

dwalkes committed:
Assignment 8 base implementation

Shared stubs from public repository and example implementation
files.

dwalkes committed:
Go to assignment 8

dwalkes committed:
Go to assignment 9

dwalkes committed:
Add aesd_ioctl.h file for assignment 9

dwalkes committed:
Update aesd_ioctl.h

Move reference to ioctl number to match latest location in the kernel source

dwalkes committed:
Merge branch 'assignment3-part-1' into assignment3-part-2

dwalkes committed:
Merge branch 'assignment3-part-2' into assignment4

dwalkes committed:
Merge branch 'assignment4' into assignment5

dwalkes committed:
Merge branch 'assignment5' into assignment6

hgajapathy committed:
Updated cross-compile.txt

Tool chain details are changed because a new cloud instance
is used from this instance.

hgajapathy committed:
Disable make and make clean

hgajapathy committed:
Updated syscalls.c

The man page of waitpid() and text book were referred for
handling child exit status.

hgajapathy committed:
Merge remote-tracking branch 'assignments-base/assignment3-part-2' into main

hgajapathy committed:
Chnage file path for assignment.txt file

hgajapathy committed:
Changed /bin/bash to /bin/sh as busybox doesnt support bash

hgajapathy committed:
Linux kernel build script

All To-Do references are added to build kernel, busybox, and
rootfs.

MELP and Lecture videos are used for references.

hgajapathy committed:
Merge remote-tracking branch 'assignments-base/assignment4' into main

Commit:
Bug fix: close open fd in do_exec_redirect()

- When the fork syscall fails, the open fd should be closed.
- As dup2(fd, 1) is redirecting command to stdout, return false
is not required. This implies a internal system call failure.
- In parent, the fd must be closed before checking child
status instead closing at the end. If any failure occurs
in child status, the code return false and the fd is left open.

hgajapathy committed:
Completed threading implemenration for assignment4p1

hgajapathy committed:
Merge branch 'main' of github.com:cu-ecen-aeld/assignments-3-and-later-hgajapathy into main

dwalkes commented on examples/systemcalls/systemcalls.c:
```diff
@@ -92,8 +127,38 @@ bool do_exec_redirect(const char *outputfile, int count, ...)
+            if (waitpid(cpid, &wstatus, 0) == -1)
+                return false;
+            if (WEXITSTATUS(wstatus) != 0)
+                return false;
```
Nice work! This implementation should work to avoid missing cleanup on va_end.  However, see suggestions in class regarding avoiding duplication of logic with the function above.

dwalkes reviewed (commented):
Nice work!

hgajapathy committed:
Modified finder-test.sh for assignment-4-part-2

hgajapathy committed:
Fixed bug on finder-test.sh

The file must contain the expanded value of OUTPUTSTRING var.

hgajapathy committed:
Fixed path name for conf/ dir

hgajapathy committed:
remove ./ from writer exe

hgajapathy committed:
Copied /tmp/assignment4-result.txt from qemu

dwalkes committed:
Merge branch 'assignment6' into assignment7

dwalkes committed:
Merge branch 'assignment7' into assignment8

Commit:
Merge remote-tracking branch 'assignments-base/assignment5' into main

hgajapathy committed:
Assignment 5 Part 1: Native Socket Server

Added aesdsocet server.

The implementation is tested with text scripts.

hgajapathy committed:
Assignment 5 Part 2: Start-up scripts

Added startup scripts to start/stop aesdsocket server

hgajapathy committed:
Merge remote-tracking branch 'assignments-base/assignment6' into main

hgajapathy committed:
Assignment 6 Part 1 Updates

A work-in-progress multithreaded socket server. The commit is
verified with full_test.sh and match "Validation" requirement
given in the assignment document.

valgrind-out.txt output for this commit

==207038==
==207038== HEAP SUMMARY:
==207038==     in use at exit: 272 bytes in 1 blocks
==207038==   total heap usage: 122 allocs, 121 frees, 180,855 bytes allocated
==207038==
==207038== Searching for pointers to 1 not-freed blocks
==207038== Checked 8,478,752 bytes
==207038==
==207038== 272 bytes in 1 blocks are possibly lost in loss record 1 of 1
==207038==    at 0x483DD99: calloc (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_memcheck-amd64-linux.so)
==207038==    by 0x40149DA: allocate_dtv (dl-tls.c:286)
==207038==    by 0x40149DA: _dl_allocate_tls (dl-tls.c:532)
==207038==    by 0x485E322: allocate_stack (allocatestack.c:622)
==207038==    by 0x485E322: pthread_create@@GLIBC_2.2.5 (pthread_create.c:660)
==207038==    by 0x10B1F3: aesdsocket (in /home/haga9942/assignments-3-and-later/server/aesdsocket)
==207038==    by 0x10B5F0: main (in /home/haga9942/assignments-3-and-later/server/aesdsocket)
==207038==
==207038== LEAK SUMMARY:
==207038==    definitely lost: 0 bytes in 0 blocks
==207038==    indirectly lost: 0 bytes in 0 blocks
==207038==      possibly lost: 272 bytes in 1 blocks
==207038==    still reachable: 0 bytes in 0 blocks
==207038==         suppressed: 0 bytes in 0 blocks
==207038==
==207038== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)

hgajapathy committed:
Assignment 6 Part 1 Updates

Match yocto linker flags

hgajapathy committed:
Makefile test updates

hgajapathy committed:
bug fix

hgajapathy committed:
Assignment 6 Part 1 Updates

Final update for aesdsocket server and Makefile to match
Yocto build.

aesdsocket server is updated with Assignment 5 comments
and change 'read' call with 'pread' to avoid lseek
function.

Valgrind output to show full_test.sh results

==2866798== HEAP SUMMARY:
==2866798==     in use at exit: 272 bytes in 1 blocks
==2866798==   total heap usage: 122 allocs, 121 frees, 191,321 bytes allocated
==2866798==
==2866798== Searching for pointers to 1 not-freed blocks
==2866798== Checked 8,478,752 bytes
==2866798==
==2866798== 272 bytes in 1 blocks are possibly lost in loss record 1 of 1
==2866798==    at 0x483DD99: calloc (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_memcheck-amd64-linux.so)
==2866798==    by 0x40149DA: allocate_dtv (dl-tls.c:286)
==2866798==    by 0x40149DA: _dl_allocate_tls (dl-tls.c:532)
==2866798==    by 0x4860322: allocate_stack (allocatestack.c:622)
==2866798==    by 0x4860322: pthread_create@@GLIBC_2.2.5 (pthread_create.c:660)
==2866798==    by 0x10B2C1: aesdsocket (aesdsocket.c:370)
==2866798==    by 0x10B6D7: main (aesdsocket.c:478)
==2866798==
==2866798== LEAK SUMMARY:
==2866798==    definitely lost: 0 bytes in 0 blocks
==2866798==    indirectly lost: 0 bytes in 0 blocks
==2866798==      possibly lost: 272 bytes in 1 blocks
==2866798==    still reachable: 0 bytes in 0 blocks
==2866798==         suppressed: 0 bytes in 0 blocks
==2866798==
==2866798== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
deleting marker file /tmp/tmp.sy6U3t0ywx
Test of assignment assignment6 complete with success

hgajapathy committed:
Assignment 6 Part 1 Updates

Commented #define DEBUG to suppress log sent to stdout

hgajapathy committed:
Assignmnet 6 Part 1 Updates

Modified Makefile to match lecture videos

hgajapathy committed:
Assignment 6 Part 1 Updates

buf in process_msg can be statically allocated instead
instead of dynamic allocation.

hgajapathy committed:
Merge remote-tracking branch 'assignments-base/assignment7' into main

hgajapathy committed:
Assignment 7 Part 1 Updates

Updated aesd-circular-buffer.c based on assignment instruction.

hgajapathy committed:
Assignment 7 Part 2 Updates

added assignments/assignment7/faulty-oops.md

hgajapathy committed:
Merge remote-tracking branch 'assignments-base/assignment8' into main

hgajapathy committed:
Assignment 8 Updates

This commit has all the updates to meet assignment 8 requirments.

hgajapathy committed:
Assignment 8 Updates

Check input arguments for NULL and destroy mutex in cleanup.

hgajapathy committed:
Merge remote-tracking branch 'assignments-base/assignment9' into main

hgajapathy committed:
Assignment 9 Updates

1. updated aesdchar driver to handle seek and ioctl command.
2. modified aesdsocket to handle AESDCHAR_IOCSEEKTO command.
